sdk/storage/azure-storage-blob/src/main/java/com/azure/android/storage/blob/StorageBlobServiceImpl.java

```diff
@@ -1839,6 +1848,18 @@ private BlockBlobsStageBlockResponse stageBlockWithRestResponseIntern(String con
         }
         //
         final String comp = "block";
+
+        if (computeMd5) {
+            if (transactionalContentMD5 != null) {
+                throw new IllegalArgumentException("'transactionalContentMD5' can not be set when 'computeMd5' is true.");
+            }
+            try {
+                transactionalContentMD5 = MessageDigest.getInstance("MD5").digest(blockContent);
+            } catch (NoSuchAlgorithmException e) {
+                throw new RuntimeException(e);
```

Comment:
I wonder why we would prefer to throw this as a `RuntimeException` instead of the original type.

Reply:
This is what we do in Java usually. I think its just a remnant of some logging deficiency that used to exist and Im still not used to keeping the original type. Let me change to just throw the original exception.

Reply:
Hmm if I change it to just throw its mad that Im not declaring the exception as part of the method type.

Reply:
Ok, let's leave it like this for now :)